electrumsv/wallet.py

```diff
@@ -3251,6 +3251,7 @@ def update_cached_values(future: concurrent.futures.Future[list[NetworkServerRow
             for specific_server_key in deleted_server_keys:
                 server = self._servers[specific_server_key.to_base_key()]
                 server.clear_server_account_usage(specific_server_key)
+                del self._servers[specific_server_key.to_base_key()]
```

Comment:
I don't think you understand this code. It is clearing the account usage for the given server entry and no deleting the server because we do not know we are no longer using the server. 

1. You cannot delete the server entry here.
2. If you are going to delete the server entry you need to make sure that all account usages have been cleared and there is no longer any use of this server.

Reply:
Okay. That makes it very complicated to delete from `self._servers`. 

Does this mean we need to check:
- `Network.connected_header_server_states`
- `Wallet._worker_tasks_maintain_server_connection`
- ? Active DPP server connections if it's a DPP type server
- ? Peer Channel or Tip Filtering cleanups

And because there is no existing mechanism to force a clean disconnection from a given server, do we need a background task that waits for a disconnection event and only deletes the server (marked for eventual deletion) when all criteria for it not being currently in use by anything are met?

Any guidance appreciated...